### PR TITLE
For #27522 new delete one/all search engines UI tests

### LINKED ISSUE #27522: [UI test automation] Improve coverage of the search test suite
### Summary
Improve coverage of the search test suite :
• Create new UI tests
• Add missing steps

### PR DESCRIPTION
**Pull request by AndiAJ**
For #27522 new delete one/all search engines UI tests.
All 3 UI tests successfully passed 100x on Firebase. ✅ 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #27522

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsSearchTest.kt [hunk 1]
```diff
@@ -220,4 +220,59 @@ class SettingsSearchTest {
...
+            verifyThreeDotButtonIsNotDisplayed("Wikipedia")
+            openAddSearchEngineMenu()
+            verifyAddSearchEngineListContains(
+                "Google",
+                "Bing",
+                "Amazon.com",
+                "DuckDuckGo",
+                "eBay",
+            )
+        }
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsSearchTest.kt:276]: Can we add a comment for each test here that these are expected `en-us` defaults?

**Review by AaronMT — APPROVED** [review]
Looks good to me. Please add a comment about these `en-US` defaults.

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsSearchTest.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SettingsSearchTest.kt:279]: Done !

**Comment by AndiAJ** [comment]
@JohanLorenzo Seems like the CI is stuck. Do you know what could be the issue?

**Comment by JohanLorenzo** [comment]
Hey @AndiAJ! Let's continue the conversation there https://github.com/mozilla-mobile/fenix/pull/27567#issuecomment-1294766679 🙂

**Comment by Mugurell** [comment]
Seems like this was stuck at being embarked in the merged train probably because of the same issue as above even after I initially run all CI tasks in my PR.
Removed the `needs-landing` label to unblock other landings.

**Comment by Mugurell** [comment]
Seems like the PR is stucked at "Queue: Embarked in merge train".
Closing the PR to unblock other merges.

**Pull request merged**